### PR TITLE
(PA-1763) Update to ruby 2.4.3; Remove ruby 2.1.9 and 2.3.5 files

### PR DESCRIPTION
**Pull request by caseywilliams**
- Removes supporting files and a README reference for ruby 2.1.9 and ruby 2.3.5
- Updates ruby 2.4.2 component to 2.4.3

### CODE REVIEW

**Comment by caseywilliams** [comment]
jenkins please test this

**Comment by caseywilliams** [comment]
jenkins please test this

**Comment by caseywilliams** [comment]
I realized I'd somehow moved one of the support files to a new location; Updated to fix that.

File: resources/files/ruby_243/rbconfig/rbconfig-i386-pc-solaris2.10.rb [hunk 1]
```diff
@@ -3,16 +3,16 @@
...
 
   TOPDIR = File.dirname(__FILE__).chomp!("/lib/ruby/2.4.0/i386-solaris2.10")
   DESTDIR = '' unless defined? DESTDIR
   CONFIG = {}
   CONFIG["DESTDIR"] = DESTDIR
   CONFIG["MAJOR"] = "2"
   CONFIG["MINOR"] = "4"
-  CONFIG["TEENY"] = "2"
+  CONFIG["TEENY"] = "3"
   CONFIG["PATCHLEVEL"] = "111"
```
Comment (geoffnichols) [inline resources/files/ruby_243/rbconfig/rbconfig-i386-pc-solaris2.10.rb]: This is a nitpick, but we might want to consider bumping PATCHLEVEL and RUBY_RELEASE per https://github.com/ruby/ruby/blob/v2_4_3/version.h#L2-L3.

**Comment by geoffnichols** [comment]
Overall this looks good to me. 👍

**Comment by caseywilliams** [comment]
jenkins please test this on windows2012r2-64,aix71-POWER

**Pull request merged**